### PR TITLE
Refactored repositories after review

### PR DESCRIPTION
**Pull request by zinkoviktor**

### CODE REVIEW

File: NetCoreRestApi/DataLayer.EF/Repositories/BaseRepository.cs [hunk 1]
```diff
@@ -12,41 +13,41 @@ public class BaseRepository<TModel, TEntity, TId> : IRepository<TModel, TId>
         where TEntity : BaseEntity<TId>
     {
         protected IConverter<TEntity, TModel> Сonverter { get; private set; }
-        protected IRepositoryDbContext<TEntity> DBContext { get; private set; }
+        protected DbSet<TEntity> DbSet { get; private set; }
```
Comment (mdemii) [inline NetCoreRestApi/DataLayer.EF/Repositories/BaseRepository.cs]: Private field

File: NetCoreRestApi/DataLayer.EF/EfDbContext.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using DataLayer.EF.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace DataLayer.EF
+{
+    public class EfDbContext<TEntity> : IRepositoryDbContext<TEntity>
```
Comment (mdemii) [inline NetCoreRestApi/DataLayer.EF/EfDbContext.cs]: Rename to EFDbContextMock

File: NetCoreRestApi/DataLayer.EF/DbSetMock.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace DataLayer.EF
+{
+    public class DbSetMock<TEntity> : DbSet<TEntity>
```
Comment (mdemii) [inline NetCoreRestApi/DataLayer.EF/DbSetMock.cs:10]: As temporary solution, it is ok. later this will be replaced with EF core dbcontext in memory

**Review by mdemii — CHANGES_REQUESTED** [review]
Minor changes required

File: NetCoreRestApi/DataLayer.EF/Repositories/BaseRepository.cs (hunk 1 above)
Comment (zinkoviktor) [reply NetCoreRestApi/DataLayer.EF/Repositories/BaseRepository.cs]: fixed

File: NetCoreRestApi/DataLayer.EF/DbSetMock.cs (hunk 3 above)
Comment (zinkoviktor) [reply NetCoreRestApi/DataLayer.EF/DbSetMock.cs:10]: ok

File: NetCoreRestApi/DataLayer.EF/EfDbContext.cs (hunk 2 above)
Comment (zinkoviktor) [reply NetCoreRestApi/DataLayer.EF/EfDbContext.cs]: fixed

**Review by mdemii — APPROVED** [review]

**Pull request merged**